pkg/local_object_storage/metabase/v2/put.go

```diff
@@ -0,0 +1,108 @@
+package meta
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/nspcc-dev/neofs-node/pkg/core/object"
+	"go.etcd.io/bbolt"
+)
+
+type (
+	namedBucketItem struct {
+		name, key, val []byte
+	}
+)
+
+// Put saves object header in metabase. Object payload expected to be cut.
+// Big objects have nil blobovniczaID.
+func (db *DB) Put(obj *object.Object, blobovniczaID *uuid.UUID) error {
+	var isParent bool // true when object header obtained from `split.Parent`
+
+	for ; obj != nil; obj, isParent = obj.GetParent(), true {
+		exists, err := db.Exists(obj.Address())
+		if err != nil {
+			return err
+		}
+
+		// most right child and split header overlap parent so we have to
+		// check if object exists to not overwrite it twice
+		if exists {
+			continue
+		}
+
+		uniqueIndexes, err := uniqueIndexes(obj, isParent, blobovniczaID)
+		if err != nil {
+			return fmt.Errorf("can' build unique indexes: %w", err)
+		}
+
+		// todo: build list indexes
+
+		// todo: build fake bucket tree indexes
+
+		// consider making one TX for both target object and parent
+		err = db.boltDB.Update(func(tx *bbolt.Tx) error {
+			// put unique indexes
+			for i := range uniqueIndexes {
+				err := putUniqueIndexItem(tx, uniqueIndexes[i])
+				if err != nil {
+					return err
+				}
+			}
+
+			// todo: put list indexes
+
+			// todo: put fake bucket tree indexes
+
+			return nil
+		})
+		if err != nil { // if tx failed then return error
+			return err
+		}
+	}
+
+	return nil
+}
+
+func uniqueIndexes(obj *object.Object,
+	isParent bool,
+	blobovniczaID *uuid.UUID) ([]namedBucketItem, error) {
+	addr := obj.Address()
+	result := make([]namedBucketItem, 0, 10)
+
+	// add value to primary unique bucket
+	if !isParent {
+		rawObject, err := obj.Marshal()
+		if err != nil {
+			return nil, fmt.Errorf("can't marshal object header: %w", err)
+		}
+
+		result = append(result, namedBucketItem{
+			name: primaryBucketName(addr.ContainerID()),
+			key:  objectKey(addr.ObjectID()),
+			val:  rawObject,
+		})
+	}
+
+	// index blobovniczaID if it is present
+	if blobovniczaID != nil {
+		result = append(result, namedBucketItem{
+			name: smallBucketName(addr.ContainerID()),
+			key:  objectKey(addr.ObjectID()),
+			val:  []byte(blobovniczaID.String()),
+		})
+	}
```

Comment:
It seems like `blobovniczaID` makes sense only for non parent objects. Probably this code should be moved up in `!isParent` branch.